toqito/nonlocal_games/binary_constraint_system_game.py

```diff
@@ -0,0 +1,202 @@
+import numpy as np
+import networkx as nx
+from toqito.nonlocal_games.nonlocal_game import NonlocalGame
+
+def create_bcs_constraints(M: np.ndarray, b: np.ndarray):
+    r"""Construct a list of constraints in tensor form for a binary constraint system (BCS) game.
+    This function builds a list of constraints by converting each row of the binary matrix
+    M of shape (m, n) and the corresponding element of the binary vector b
+    into an n-dimensional tensor of shape (2, 2, ..., 2) (one axis per variable).
+    The conversion works as follows:
+    
+      1. For the i-th constraint, compute the constant value as 
+         rhs = (-1)**(b[i]).
+      2. Create an n-dimensional array (tensor) of shape (2,)*n filled with -rhs.
+      3. Compute the index from the first n entries of the i-th row of M by taking each value modulo 2.
+      4. Set the tensor element at that index to rhs.
+    For example, if 
+       M[i] = [1, 1] and b[i] = 0 (so rhs = 1),
+    then the tensor is of shape (2, 2) and is created as follows:
+    
+      - Start with a (2, 2) array filled with -1 (since -rhs = -1):
+            [ [-1, -1],
+              [-1, -1] ]
+      - The index is computed as (1 % 2, 1 % 2) = (1, 1).
+      - At position (1, 1), the value is set to 1, resulting in:
+            [ [-1, -1],
+              [-1,  1] ]
+    This tensor now represents the constraint in full detail.
+    Examples
+    ==========
+        >>> import numpy as np
+        >>> from binary_constraint_system_game import create_bcs_constraints
+        >>> M = np.array([[1, 1], [1, 1]], dtype=int)
+        >>> b = np.array([0, 1], dtype=int)
+        >>> constraints = create_bcs_constraints(M, b)
+        >>> constraints[0].shape
+        (2, 2)
+    References
+    ============
+        (See bibliography in relevant documentation)
+    :param M: A binary matrix of shape (m, n) defining which variables appear in each constraint.
+    :param b: A binary vector of length m that determines the constant term (-1)**(b[i]).
+    :return: A list of n-dimensional NumPy arrays (tensors) of shape ((2,)*n) representing each constraint.
+    """
+    m, n = M.shape
+    constraints = []
+    for i in range(m):
+        rhs = (-1) ** b[i]
+        # Create an n-dimensional array of shape (2, 2, ..., 2) filled with -rhs.
+        constraint_tensor = np.full((2,) * n, fill_value=-rhs, dtype=int)
+        # Compute the index from the binary row M[i]. Assume M[i] contains binary values (0 or 1).
+        idx = tuple(M[i] % 2)
+        # Set the tensor element at that index to rhs.
+        constraint_tensor[idx] = rhs
+        constraints.append(constraint_tensor)
+    return constraints
+
+
+def tensor_to_raw(constraint_tensor: np.ndarray) -> np.ndarray:
```

Comment:
I think this function more appropriately belongs under the `matrix_ops/` directory. I'm also wondering whether there is a more descriptive and perhaps more widely understood name for this function. Perhaps `tensor_inverse` or `tensor_unravel`? 

Additionally, we should do two additional things here:

1. Unit tests to this function explicitly in `matrix_ops/tests` and
2. The docstring example should make use of `.. jupyter-execute` (refer to any of the docstrings in any of the other examples that use this convention for an example). 

Also, it seems like this only works for matrices that have two unique inputs. Is there some well-defined function idea or name that describes this operation? I'm not aware of any myself, but I could just be unaware.